faunadb-scala/src/main/scala/faunadb/FaunaClient.scala

```diff
@@ -190,11 +196,44 @@ class FaunaClient private (connection: Connection) {
     response
       .flatMap {
         case successResponse if successResponse.statusCode() < 300 => handleSuccessResponse(successResponse)
-        case errorResponse => handleErrorResponse(errorResponse)
+        case errorResponse => handleErrorResponse(errorResponse.statusCode(), errorResponse.body())
```

Comment:
We can reuse all the error handling infra. by not making it aware of the full `Response` object.
This is validated with 2 unit tests.